code/modules/food_and_drinks/food/snacks_other.dm

```diff
@@ -531,6 +531,33 @@
 	tastes = list("paint thinner" = 1)
 	color = "#EE35FF"
 
+/obj/item/reagent_containers/food/snacks/chewable/bubblegum/bubblegum
+	name = "bubblegum gum"
+	desc = "A rubbery strip of gum. You don't feel like eating it is a good idea."
+	color = "#913D3D"
+	list_reagents = list(/datum/reagent/blood = 15)
+	tastes = list("hell" = 1)
+
+/obj/item/reagent_containers/food/snacks/chewable/bubblegum/bubblegum/process()
+	..()
+	if(iscarbon(loc))
+		hallucinate(loc)
+
+
+/obj/item/reagent_containers/food/snacks/chewable/bubblegum/bubblegum/On_Consume(mob/living/eater)
+	. = ..()
+	if(iscarbon(eater))
+		hallucinate(eater)
+
+///This proc has a 5% chance to have a bubblegum line appear, with an 85% chance for just text and 15% for a bubblegum hallucination and scarier text.
+/obj/item/reagent_containers/food/snacks/chewable/bubblegum/bubblegum/proc/hallucinate(mob/living/carbon/victim)
+	if(prob(5)) //cursed by bubblegum
```

Comment:
Early return.

Reply:
how am i supposed to get an early return there

Reply:
if(!prob(5))

Reply:
whats the difference between returning when 5% chance fails and doing stuff if 5% chance happens

Reply:
early returns are healthy and serve to reduce the amount of indentations
but do if(!prob(95)) instead of what qust said because that'd still be the same 5% chance of ocurring

Reply:
What Qust said meant
`if(!prob(5)) return`
Alternatively `if(prob(95)) return` would work the same.

On early returns, they are part of the contribution guidelines:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-early-return